From 4077e95bb16c3331220b73a1d0cdbc960789c3dc Mon Sep 17 00:00:00 2001
From: Bai Ping <ping.bai@nxp.com>
Date: Mon, 24 Apr 2017 18:45:17 +0800
Subject: [PATCH 2533/5242] MLK-14747 driver: cpufreq: Correct dc regulator
 voltage on imx6ull

commit  f6ed36d922180c33b0bc0bacbaab1e3e2a382278 from
https://source.codeaurora.org/external/imx/linux-imx.git

On i.MX6ULL EVK board, when the overdrive mode(900MHz/800MHz) is
enable, the DC regulator voltage should not be changed. Keep the
DC regulator to default 1.4V.

Signed-off-by: Bai Ping <ping.bai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/cpufreq/imx6q-cpufreq.c |    9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/drivers/cpufreq/imx6q-cpufreq.c b/drivers/cpufreq/imx6q-cpufreq.c
index 01c13a2..0a8d3e8 100644
--- a/drivers/cpufreq/imx6q-cpufreq.c
+++ b/drivers/cpufreq/imx6q-cpufreq.c
@@ -1,5 +1,6 @@
 /*
- * Copyright (C) 2013 Freescale Semiconductor, Inc.
+ * Copyright (C) 2013-2016 Freescale Semiconductor, Inc.
+ * Copyright 2017 NXP.
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -25,7 +26,7 @@
 #define DC_VOLTAGE_MAX		1400000
 #define FREQ_1P2_GHZ		1200000000
 #define FREQ_396_MHZ		396000
-#define FREQ_696_MHZ		696000
+#define FREQ_528_MHZ		528000
 #define FREQ_198_MHZ		198000
 #define FREQ_24_MHZ		24000
 
@@ -519,10 +520,10 @@ static int imx6q_cpufreq_probe(struct platform_device *pdev)
 	}
 
 	/*
-	 * On i.MX6UL EVK board, if the SOC is run in overide frequency,
+	 * On i.MX6UL/ULL EVK board, if the SOC is run in overide frequency,
 	 * the dc_regulator voltage should not be touched.
 	 */
-	if (freq_table[num - 1].frequency == FREQ_696_MHZ)
+	if (freq_table[num - 1].frequency > FREQ_528_MHZ)
 		ignore_dc_reg = true;
 	if (!IS_ERR(dc_reg) && !ignore_dc_reg)
 		regulator_set_voltage_tol(dc_reg, DC_VOLTAGE_MIN, 0);
-- 
1.7.9.5

